packages/modules/e3dc/inverter.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+from typing import Dict, Union
+import logging
+
+from dataclass_utils import dataclass_from_dict
+from modules.common import modbus
+from modules.common.component_state import InverterState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.modbus import ModbusDataType, Endian
+from modules.common.fault_state import ComponentInfo
+from modules.common.store import get_inverter_value_store
+from modules.common.simcount._simcounter import SimCounter
+from modules.e3dc.config import E3dcInverterSetup
+
+
+log = logging.getLogger(__name__)
+
+
+def read_inverter(client: modbus.ModbusTcpClient_, read_ext) -> [int, int]:
+    # 40067 PV Leistung
+    pv = client.read_holding_registers(40067, ModbusDataType.INT_32,
+                                       wordorder=Endian.Little, unit=1) * -1
+    if read_ext == 1:
+        # 40075 externe PV Leistung
+        pv_external = client.read_holding_registers(40075, ModbusDataType.INT_32,
+                                                    wordorder=Endian.Little, unit=1)
+    else:
+        pv_external = 0
+    return pv, pv_external
+
+
+class E3dcInverter:
+    def __init__(self,
+                 device_id: int,
+                 component_config: Union[Dict, E3dcInverterSetup]) -> None:
+        self.__device_id = device_id
+        self.component_config = dataclass_from_dict(E3dcInverterSetup, component_config)
+        self.sim_counter = SimCounter(self.__device_id, self.component_config.id, prefix="pv")
```

Comment:
```suggestion
        self.component_config = dataclass_from_dict(E3dcInverterSetup, component_config)
        self.sim_counter = SimCounter(device_id, self.component_config.id, prefix="pv")
```
Da `self.__device_id` sonst nie gelesen wird...

Reply:
geändert